bofire/strategies/predictives/qparego.py

```diff
@@ -78,18 +79,15 @@ def _get_objective(
             weights=weights, Y=obj_callable(preds, None) * ref_point_mask
         )
 
-        def objective(Z, X=None):
+        def objective_callable(Z, X=None):
             return scalarization(obj_callable(Z, None) * ref_point_mask, X)
 
         if len(weights) != len(self.domain.outputs):
```

Comment:
```suggestion
        if len(weights) != len(self.domain.outputs.get_by_objective(Objective)):
```

This is a bug from my side, which can lead in edge cases to problems.

Reply:
This has been changed accordingly.